bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -169,7 +168,7 @@ public String toString()
     return identity;
   }
 
-  private final HashMap<String, DataList> publisherBuffers = new HashMap<String, DataList>();
+  private final ConcurrentHashMap<String, DataList> publisherBuffers = new ConcurrentHashMap<String, DataList>();
```

Comment:
Please use 1 for concurrency level instead of default 16. There is only one thread that modifies publisherBuffers. Use default 0.75f for the load factor and 1 as the initial capacity.